Unused variables found in the SHM test

Found this issue while testing [setup-rust-toolchain](https://github.com/actions-rust-lang/setup-rust-toolchain)
that its default `RUSTFLAGs=-D warnings` caused the following test to fail.

```bash
cargo nextest run -F shared-memory -F transport_unixpipe -p zenoh-transport
```

Error logs

```bash
error: unused import: `Priority`
Error:   --> io/zenoh-transport/tests/multicast_transport.rs:31:61
   |
31 |             Channel, CongestionControl, Encoding, EndPoint, Priority, Reliability, WhatAmI, ZenohId,
   |                                                             ^^^^^^^^
   |
   = note: `-D unused-imports` implied by `-D warnings`

error: constant `TIMEOUT` is never used
Error:   --> io/zenoh-transport/tests/multicast_transport.rs:48:11
   |
48 |     const TIMEOUT: Duration = Duration::from_secs(60);
   |           ^^^^^^^
   |
   = note: `-D dead-code` implied by `-D warnings`

error: constant `SLEEP` is never used
Error:   --> io/zenoh-transport/tests/multicast_transport.rs:49:11
   |
49 |     const SLEEP: Duration = Duration::from_secs(1);
   |           ^^^^^

error: constant `SLEEP_COUNT` is never used
Error:   --> io/zenoh-transport/tests/multicast_transport.rs:50:11
   |
50 |     const SLEEP_COUNT: Duration = Duration::from_millis(10);
   |           ^^^^^^^^^^^

error: constant `MSG_COUNT` is never used
Error:   --> io/zenoh-transport/tests/multicast_transport.rs:52:11
   |
52 |     const MSG_COUNT: usize = 1_000;
   |           ^^^^^^^^^

error: constant `MSG_SIZE_NOFRAG` is never used
Error:   --> io/zenoh-transport/tests/multicast_transport.rs:53:11
   |
53 |     const MSG_SIZE_NOFRAG: [usize; 1] = [1_024];
   |           ^^^^^^^^^^^^^^^

error: method `get_count` is never used
Error:   --> io/zenoh-transport/tests/multicast_transport.rs:69:12
   |
68 |     impl SHPeer {
   |     ----------- method in this implementation
69 |         fn get_count(&self) -> usize {
   |            ^^^^^^^^^

error: struct `TransportMulticastPeer` is never constructed
Error:    --> io/zenoh-transport/tests/multicast_transport.rs:134:12
    |
134 |     struct TransportMulticastPeer {
    |            ^^^^^^^^^^^^^^^^^^^^^^

error: function `open_transport` is never used
Error:    --> io/zenoh-transport/tests/multicast_transport.rs:140:14
    |
140 |     async fn open_transport(
    |              ^^^^^^^^^^^^^^

error: function `close_transport` is never used
Error:    --> io/zenoh-transport/tests/multicast_transport.rs:226:14
    |
226 |     async fn close_transport(
    |              ^^^^^^^^^^^^^^^

error: function `test_transport` is never used
Error:    --> io/zenoh-transport/tests/multicast_transport.rs:250:14
    |
250 |     async fn test_transport(
    |              ^^^^^^^^^^^^^^

error: function `run_single` is never used
Error:    --> io/zenoh-transport/tests/multicast_transport.rs:302:14
    |
302 |     async fn run_single(endpoint: &EndPoint, channel: Channel, msg_size: usize) {
    |              ^^^^^^^^^^

error: function `run` is never used
Error:    --> io/zenoh-transport/tests/multicast_transport.rs:317:14
    |
317 |     async fn run(endpoints: &[EndPoint], channel: &[Channel], msg_size: &[usize]) {
    |              ^^^

error: could not compile `zenoh-transport` (test \"multicast_transport\") due to 13 previous errors
warning: build failed, waiting for other jobs to finish...
error: command `/home/runner/.rustup/toolchains/1.72.0-x86_64-unknown-linux-gnu/bin/cargo test --no-run --message-format json-render-diagnostics --package zenoh-transport --features shared-memory --features transport_unixpipe` exited with code 101
Error: Process completed with exit code 101.

--- a/io/zenoh-transport/tests/multicast_transport.rs
+++ b/io/zenoh-transport/tests/multicast_transport.rs
@@ -15,6 +15,7 @@
 // Restricting to macos by default because of no IPv6 support
 // on GitHub CI actions on Linux and Windows.
 #[cfg(target_family = "unix")]
+#[cfg(all(feature = "transport_compression", feature = "transport_udp"))]
 mod tests {
     use std::{
         any::Any,

--- a/io/zenoh-transport/tests/unicast_openclose.rs
+++ b/io/zenoh-transport/tests/unicast_openclose.rs
@@ -24,6 +24,7 @@ use zenoh_transport::{
 };
 
 #[cfg(target_os = "linux")]
+#[cfg(any(feature = "transport_tcp", feature = "transport_udp"))]
 use zenoh_util::net::get_ipv4_ipaddrs;
 
 const TIMEOUT: Duration = Duration::from_secs(60);

--- a/io/zenoh-transport/tests/unicast_transport.rs
+++ b/io/zenoh-transport/tests/unicast_transport.rs
@@ -222,6 +222,11 @@ const SLEEP_COUNT: Duration = Duration::from_millis(10);
 const MSG_COUNT: usize = 1_000;
 const MSG_SIZE_ALL: [usize; 2] = [1_024, 131_072];
 const MSG_SIZE_LOWLATENCY: [usize; 2] = [1_024, 65000];
+#[cfg(any(
+    feature = "transport_tcp",
+    feature = "transport_udp",
+    feature = "transport_unixsock-stream",
+))]
 const MSG_SIZE_NOFRAG: [usize; 1] = [1_024];
 
 // Transport Handler for the router

